spinnaker/spinnaker_system/google_http_lb_upsert_scenario.py

```diff
@@ -181,62 +181,63 @@ def _add_contract_clauses(self, contract_builder, upsert):
+          'requestPath': jp.STR_EQ(hc['requestPath']),
+          'port': jp.NUM_EQ(hc['port'])
+          })
+
+    bs_clause_builder = (contract_builder
+      .new_clause_builder('Backend Services Created', retryable_for_secs=30)
+      .list_resource('backendServices')
       )
-
-    bs_clause_builder = (contract_builder.
-                         new_clause_builder('Backend Services Created',
-                                            retryable_for_secs=30).
-                         list_resource('backendServices'))
     for bs in backend_services:
-      bs_clause_builder.contains_pred_list(
-        [
-         jp.PathEqPredicate('name', bs['name']),
-         jp.PathEqPredicate('portName', 'http'),
-         jp.PathContainsPredicate('healthChecks[0]',
-                                  self._get_hc_link(bs['healthCheck']['name']))
-        ]
-      )
+      bs_clause_builder.contains_match({
+          'name': jp.STR_EQ(bs['name']),
+          'portName': jp.STR_EQ('http'),
+          'healthChecks':
+            jp.LIST_EQ([self._get_hc_link(bs['healthCheck']['name'])])
+          })
 
     url_map_clause_builder = (contract_builder
-                              .new_clause_builder('Url Map Created',
-                                                  retryable_for_secs=30)
-                              .list_resource('urlMaps'))
+      .new_clause_builder('Url Map Created', retryable_for_secs=30)
+      .list_resource('urlMaps')
+      )
+
     for hr in host_rules:
-      pred_list = []
       pm = hr['pathMatcher']
-      pred_list.append(jp.AND([
-        jp.PathEqPredicate('name', self.__lb_name),
-        jp.PathEqPredicate('defaultService',
-                           self._get_bs_link(upsert['defaultService']['name'])),
-        jp.PathContainsPredicate(
-          'pathMatchers/defaultService',
-          self._get_bs_link(pm['defaultService']['name'])),
-      ]))
-      pred_list.append(
-        jp.AND([jp.PathContainsPredicate('hostRules/hosts',
-                                         host) for host in hr['hostPatterns']])
-      )
+      path_rules_specs = []
       for pr in pm['pathRules']:
-        pred_list.append(
-         jp.PathContainsPredicate(
-           'pathMatchers/pathRules/service',
-           self._get_bs_link(pr['backendService']['name'])),
-        )
-        for path in pr['paths']:
-          pred_list.append(
-            jp.PathContainsPredicate('pathMatchers/pathRules/paths', path),
-          )
-      url_map_clause_builder.contains_pred_list(pred_list)
+        path_rules_specs.append({
+            'service': jp.STR_SUBSTR(pr['backendService']['name']),
+            'paths': jp.LIST_MATCHES([jp.STR_EQ(path) for path in pr['paths']])
+            })
+
+      url_map_clause_builder.contains_match({
+         'name': jp.STR_EQ(self.__lb_name),
+         'defaultService':
+             jp.STR_EQ(self._get_bs_link(upsert['defaultService']['name'])),
+         'pathMatchers': jp.LIST_MATCHES([
```

Comment:
Agreed. This is what I was refering to in the PR note when I said expressing lists vs elements is still awkward.